nipype/pipeline/plugins/multiproc.py

```diff
@@ -6,21 +6,63 @@
 http://stackoverflow.com/a/8963618/1183453
 """
 
+# Import packages
 from multiprocessing import Process, Pool, cpu_count, pool
 from traceback import format_exception
 import sys
 
+import numpy as np
+from copy import deepcopy
+from ..engine import MapNode
+from ...utils.misc import str2bool
+import psutil
+from ... import logging
+import semaphore_singleton
 from .base import (DistributedPluginBase, report_crash)
 
+# Init logger
+logger = logging.getLogger('workflow')
 
+# Run node
 def run_node(node, updatehash):
+    """Function to execute node.run(), catch and log any errors and
+    return the result dictionary
+
+    Parameters
+    ----------
+    node : nipype Node instance
+        the node to run
+    updatehash : boolean
+        flag for updating hash
+
+    Returns
+    -------
+    result : dictionary
+        dictionary containing the node runtime results and stats
+    """
+ 
+    # Import packages
+    import datetime
+ 
+    # Init variables
     result = dict(result=None, traceback=None)
+
+    # 
     try:
-        result['result'] = node.run(updatehash=updatehash)
+        start = datetime.datetime.now()
```

Comment:
This is already calculated here: https://github.com/nipy/nipype/blob/47587b97c99025bc72a42125fc640521b44a3431/nipype/interfaces/base.py#L1035 and written to reports here: https://github.com/nipy/nipype/blob/1262955492fa63d3d84795683fd6cc5063f0aacf/nipype/pipeline/engine/nodes.py#L741. Why not we extend this mechanism and keep everything (duration, memory, threads) in one place (reports) instead of writing new JSON file in `callback_log`?